src/services/arweave/SimpleArweaveDataProvider.ts

```diff
@@ -98,6 +101,54 @@ export class SimpleArweaveDataProvider implements ArweaveDataProvider {
     });
   }
 
+  async validateArweaveAddress(address: string): Promise<undefined | boolean> {
+    const targetAddress = new ArweaveTransactionID(address);
+    const transaction = await this._arweave.api
+      .get(`/tx/${targetAddress.toString()}`)
+      .then((res: any) => (res.status === 200 ? res : false));
+
+    if (transaction) {
+      const tags = tagsToObject(transaction.data.tags);
+      console.log(transaction);
+      const isContract = Object.values(tags).includes('SmartWeaveContract');
+      if (isContract) {
+        throw new Error(
+          `Provided address (${targetAddress.toString()}) is a SmartWeave Contract.`,
+        );
+      }
+      throw new Error(
+        `Provided address (${targetAddress.toString()} is a transaction ID and not a Smartweave Contract or an address.`,
+      );
+    }
+
+    const balance = await this._arweave.api
+      .get(`/wallet/${targetAddress.toString()}/balance`)
+      .then((res: any) => (res.status === 200 ? res.data > 0 : false));
```

Comment:
you can do `Promise.all` so these calls get called concurrently 

```
const txPromise =  this._arweave.api
      .get(`/tx/${targetAddress.toString()}`)
      .then((res: any) => (res.status === 200 ? res : false));
const balancePromise = this._arweave.api
      .get(`/wallet/${targetAddress.toString()}/balance`)
      .then((res: any) => (res.status === 200 ? res.data > 0 : false));
const gqlPromise = this._arweave.api
        .post(`/graphql`, transactionByOwnerQuery(targetAddress))
        .then((res: any) =>
          res.status === 200 ? res.data.data.transactions.edges : false,
        );


const [isTransaction, balance, hasTransactions] = await Promise.all([txPromise, balancePromise, gqlPromise]);
```

Reply:
added